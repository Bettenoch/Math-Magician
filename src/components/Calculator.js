import React, { useState } from 'react';
import '../styles/Calculator.css';
import calculate from '../logic/Calculate';

const Calculator = () => {
  const [inputData, setInputData] = useState({
    total: null,
    next: null,
    operation: null,
  });
  const handleCalc = (e) => {
    const btn = e.target.name;
    return setInputData((obj) => calculate(obj, btn));
  };
  const { total, next, operation } = inputData;
  const result = `${total}${operation}${next}`.replace(/null/g, '').replace(/undefined/g, '');

  return (
    <>
      <div className="main-container">
        <div>
          <h3>Let&apos;s do some math!!</h3>
          <div className="magician" />
        </div>
        <div className="container">
          <article className="display-content">
            <input
              type="text"
              className="input-text"
              id="calc-screen"
              value={result === '' ? '0' : result}
              disabled
            />
          </article>
          <article className="math-functions">
            <article className="math-row row-first">
              <button type="button" name="AC" onClick={handleCalc}>AC</button>
              <button type="button" name="+/-" onClick={handleCalc}>+/-</button>
              <button type="button" name="%" onClick={handleCalc}>%</button>
              <button type="button" name="รท" onClick={handleCalc}>รท</button>
            </article>

            <article className="math-row row-second">
              <button type="button" name="7" onClick={handleCalc}>7</button>
              <button type="button" name="8" onClick={handleCalc}>8</button>
              <button type="button" name="9" onClick={handleCalc}>9</button>
              <button type="button" name="x" onClick={handleCalc}>x</button>
            </article>

            <article className="math-row row-third">
              <button type="button" name="4" onClick={handleCalc}>4</button>
              <button type="button" name="5" onClick={handleCalc}>5</button>
              <button type="button" name="6" onClick={handleCalc}>6</button>
              <button type="button" name="-" onClick={handleCalc}>-</button>
            </article>

            <article className="math-row row-fourth">
              <button type="button" name="1" onClick={handleCalc}>1</button>
              <button type="button" name="2" onClick={handleCalc}>2</button>
              <button type="button" name="3" onClick={handleCalc}>3</button>
              <button type="button" name="+" onClick={handleCalc}>+</button>
            </article>

            <article className="math-row row-fifth">
              <div>
                <button type="button" name="0" onClick={handleCalc}>0</button>
              </div>
              <div>
                <button type="button" name="." onClick={handleCalc}>.</button>
                <button type="button" name="=" onClick={handleCalc}>=</button>
              </div>
            </article>
          </article>
        </div>
      </div>
    </>
  );
};

export default Calculator;
