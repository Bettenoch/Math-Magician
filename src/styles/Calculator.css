.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 5rem;
  width: 30rem;
}

.display-content {
  display: flex;
  justify-content: flex-end;
  width: 100%;
  background-color: rgb(92, 85, 85);
  padding-block: 0.8rem;
}

.input-text {
  display: flex;
  text-align: right;
  padding-right: 1rem;
  font-size: 1rem;
  background: none;
  color: aliceblue;
  outline: none;
  border: none;
}

.math-functions {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.math-row {
  display: flex;
  width: 100%;
  justify-content: space-between;
}

button {
  width: 100%;
  font-size: 1.2rem;
  height: 3rem;
  padding: 0.5rem;
  background-color: rgb(225, 209, 209);
  border: 1px solid rgb(207, 196, 196);
}

.math-row button:last-child {
  background-color: orange;
}

.math-row:last-child div {
  display: flex;
  width: 100%;
  justify-content: space-between;
}

.math-row:last-child div:first-child button {
  background-color: rgb(225, 209, 209);
}
