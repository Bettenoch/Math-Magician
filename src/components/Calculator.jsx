import React from 'react';
import '../styles/Calculator.css';
import Display from './Display';

const Calculator = () => (
  <div className="container">
    <Display className="display-content" />
    <article className="math-functions">
      <article className="math-row row-first">
        <button type="submit">AC</button>
        <button type="submit">+/-</button>
        <button type="submit">%</button>
        <button type="submit">รท</button>
      </article>

      <article className="math-row row-second">
        <button type="submit">7</button>
        <button type="submit">8</button>
        <button type="submit">9</button>
        <button type="submit">ร</button>
      </article>

      <article className="math-row row-third">
        <button type="submit">4</button>
        <button type="submit">5</button>
        <button type="submit">6</button>
        <button type="submit">-</button>
      </article>

      <article className="math-row row-fourth">
        <button type="submit">1</button>
        <button type="submit">2</button>
        <button type="submit">3</button>
        <button type="submit">+</button>
      </article>

      <article className="math-row row-fifth">
        <div>
          <button type="submit">0</button>
        </div>
        <div>
          <button type="submit">.</button>
          <button type="submit">=</button>
        </div>
      </article>
    </article>
  </div>
);

export default Calculator;
