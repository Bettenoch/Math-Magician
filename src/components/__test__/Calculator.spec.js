import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

describe('test calculator buttons', () => {
  it('should clear the items in the calculator output and give 0', () => {
    // render component in the virtual dom
    render(<Calculator />);

    // select elements to interact with user
    const clear = screen.getByTestId('clear');
    const val = screen.getByTestId('output');

    // interact with buttons

    fireEvent.click(clear);

    expect(val).toHaveValue('0');
  });
  it('should match the text content', () => {
    render(<Calculator />);
    const modulo = screen.getByTestId('modulo');
    expect(modulo).toHaveTextContent('%');
  });
});

describe('Calculator snapshots', () => {
  it('should match the calculator DOM snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
