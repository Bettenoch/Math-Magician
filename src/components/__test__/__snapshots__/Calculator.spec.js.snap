// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator snapshots should match the calculator DOM snapshot 1`] = `
<div
  className="main-container"
>
  <div>
    <h3
      data-testid="welcome-title"
    >
      Let's do some math!!
    </h3>
    <div
      className="magician"
    />
  </div>
  <div
    className="container"
  >
    <article
      className="display-content"
    >
      <input
        className="input-text"
        data-testid="output"
        disabled={true}
        id="calc-screen"
        type="text"
        value="0"
      />
    </article>
    <article
      className="math-functions"
    >
      <article
        className="math-row row-first"
      >
        <button
          data-testid="clear"
          name="AC"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          data-testid="add"
          name="+/-"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          data-testid="modulo"
          name="%"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          name="รท"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </article>
      <article
        className="math-row row-second"
      >
        <button
          name="7"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          name="8"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          name="9"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          name="x"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </article>
      <article
        className="math-row row-third"
      >
        <button
          name="4"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          name="5"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          name="6"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          name="-"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </article>
      <article
        className="math-row row-fourth"
      >
        <button
          name="1"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          name="2"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          name="3"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          name="+"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </article>
      <article
        className="math-row row-fifth"
      >
        <div>
          <button
            name="0"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
        </div>
        <div>
          <button
            name="."
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            name="="
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </article>
    </article>
  </div>
</div>
`;
